listenbrainz/webserver/views/profile.py

```diff
@@ -123,6 +124,39 @@ def import_data():
     )
 
 
+def fetch_listens(musicbrainz_id, to_ts):
+    """
+    Fetch all listens for the user from listenstore by making repeated queries
+    to listenstore until we get all the data. Returns a generator that streams
+    the results.
+    """
+    db_conn = webserver.create_influx(current_app)
+    while True:
+        batch = db_conn.fetch_listens(current_user.musicbrainz_id, to_ts=to_ts, limit=EXPORT_FETCH_COUNT)
+        if not batch:
+            break
+        yield from batch
+        to_ts = batch[-1].ts_since_epoch  # new to_ts will be the the timestamp of the last listen fetched
+
+
+def map_listen_for_export(obj):
```

Comment:
we have a function for this (https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/listen.py#L186) that might be helpful.

Reply:
Thanks for taking a look! `map_listen_for_export` here is what used to be [this loop body](https://github.com/metabrainz/listenbrainz-server/pull/708/files#diff-5c1b6caf808da8edd5873fe0fc2e5d22L147-L152). It looks like `to_api` returns the data in a slightly different format. If we use it here, that would change the schema of `/profile/export`. Is that desired?

Reply:
I think so, yes! We're hoping to consolidate the data format. The old version seems to be from some old code that never got changed.

Reply:
Okay, I changed it to use `to_api` and I updated the test to match the new format.